CRM/Core/Payment/AuthorizeNet.php

```diff
@@ -194,12 +202,10 @@ public function doDirectPayment(&$params) {
         // Success
         $params['trxn_id'] = !empty($response_fields[6]) ? $response_fields[6] : $this->getTestTrxnID();
         $params['gross_amount'] = $response_fields[9];
+        $result['payment_status_id'] = array_search('Completed', $contributionStatus);
         break;
     }
-
-    // TODO: include authorization code?
-
-    return $params;
+    return $result;
```

Comment:
@eileenmcnaughton We are treating $result/$params as an array but it will actually get returned as a PropertyBag if it is passed in as one. I think we should explicitly be returning an array from doPayment and it should only contain a specific set of parameters.

Reply:
@mattwire - yes, I agree - I was waiting on the other one to be merged before making that change anywhere else but it was the reason I used a separate array & didn't add to params

Reply:
The test fail related -I've fixed it & commented it heavily - I can't stop setting trxn_id in params until we've resolved https://github.com/civicrm/civicrm-core/pull/18150 & done the follow up by checking out the other forms - but I've at least commented it